### PR TITLE
Add Sequence::fill method

### PR DESCRIPTION
**Pull request by castarco**
Added `Sequence::fill` method as commented in ( https://github.com/php-ds/extension/issues/71 ).

### CODE REVIEW

File: src/Sequence.php [hunk 1]
```diff
@@ -99,6 +99,27 @@ public function apply(callable $callback)
...
     /**
+     * @param mixed $value
+     * @param null|int $num
+     */
+    public function fill($value, int $num = null)
+    {
+        if (null === $num) {
+            \array_walk($this->array, function (&$v) use ($value) {
+                $v = $value;
+            });
```
Comment (rtheunissen) [inline src/Sequence.php]: Isn't this just: `$this->array = array_fill(0, count($this), $value);`? I don't think we need `array_walk` and reference here.

File: src/Sequence.php [hunk 2]
```diff
@@ -99,6 +99,27 @@ public function apply(callable $callback)
...
+        } else {
+            if ([] === $this->array) {
+                $this->array = \array_fill(0, $num, $value);
+            } else {
+                for ($i = 0; $i < $num; $i++) {
+                    $this->array[$i] = $value;
+                }
+            }
```
Comment (rtheunissen) [inline src/Sequence.php]: For simplicity (and the fact that performance isn't critical for the polyfill), I think we can skip the empty array check and only run the loop.

File: src/Sequence.php [hunk 3]
```diff
@@ -99,6 +99,27 @@ public function apply(callable $callback)
     }
 
     /**
+     * @param mixed $value
+     * @param null|int $num
+     */
```
Comment (castarco) [inline src/Sequence.php]: Which exception should be thrown? DomainException? or LengthException?

**Comment by castarco** [comment]
I've applied your suggestions.

**Comment by rtheunissen** [comment]
✨

**Pull request merged**